### DISCUSSION TITLE
ARM Docker images?

### DISCUSSION THREAD

**Discussion by ismailkarsli** [aleksilassila/reiverr#82] [Q&A] [answered]
Is it possible to run this app on ARM processor?

**Comment by aleksilassila** [comment]
Yes, but you'll have to either build a docker image from source or just run the x64 images, which may work on some arm machines but is not ideal

**Reply by ismailkarsli** [reply]
Would [this line](https://github.com/aleksilassila/reiverr/blob/addbe36093266058a0eb32f3776027abb3aad9ae/Dockerfile#L15C7-L15C7) could be issue? it looks like hard-coded amd64.

Is it not possible you to change the Github workflow? There's an [example here](https://github.com/Xyphyn/photon/blob/main/.github/workflows/docker-image.yml). I can try it on a fork and request a PR if you want.

Building the image locally is not an ideal solution since I will need to rebuild every release.

**Reply by aleksilassila** [reply]
I agree, it's not ideal. That line is there to allow me to build x64 using my arm laptop, but now that I think about it I don't have to do that since the github workflow automates building and publishing the images. 

On a glance the example repository you provided does seem to have the setup I'm after: the images are created when a new release is created and tagged with both latest and version. PR would be really helpful, if you know how to get the multi-platform builds working!

**Reply by ismailkarsli** [reply]
I really can't say I'm experienced in this but let me try and see :) Btw I'm on ARM too and I can build with the command below:
```docker buildx --platform linux/amd64,linux/arm64 .```

**Reply by ismailkarsli** [reply, marked as answer]
Can you check the PR? @aleksilassila 
https://github.com/aleksilassila/reiverr/pull/83

**Reply by aleksilassila** [reply]
👍 I'll check when i get home

**Reply by bjornlll** [reply]
This might be old news to most, but for those that are here and are looking for a temp fix to get it working on ARM:

1. Clone this repository to a local folder somewhere
2. Open up the Dockerfile and replace `linux/amd64` with `linux/arm64/v8`
3. Build the dockerfile: `docker build -t reiverr-arm .`
4. Reference the newly build image in your docker compose file (if applicable)

Love the UI. Hope this project lives on!
